### PR TITLE
Remove unnecessary interop from System.Net.NetworkInformation

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/32944
cc: @davidsh, @jp2masa

### CODE REVIEW

File: src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Windows.cs [hunk 1]
```diff
@@ -238,23 +230,20 @@ private static void StartHelper(NetworkAddressChangedEventHandler caller, bool c
 
                     if (!s_isPending)
                     {
-                        int length;
-                        SocketError errorCode;
-
                         if (Socket.OSSupportsIPv4 && (startIPOptions & StartIPOptions.StartIPv4) != 0)
                         {
-                            s_registeredWait = ThreadPool.RegisterWaitForSingleObject(
+                            ThreadPool.RegisterWaitForSingleObject(
```
Comment (davidsh) [inline src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Windows.cs]: Can you remind me why we don't care about saving this field?

**Comment by davidsh** [comment]
@dotnet-bot Test this please

**Comment by jp2masa** [comment]
Not sure if it's in the scope of this PR, but I think that `SafeSocketHandle` should be moved from the common folder to `System.Net.Sockets`. Also, not sure if `SafeCloseSocketAndEvent` is used anywhere else, I think it can be removed.

File: src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Windows.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Windows.cs]: We never do anything with it :)  It's only benefit would be unregistering, and we're not doing that.  Further, we're actually overwriting the IPv4 one with the IPv6, so obviously it's not needed ;)

**Comment by stephentoub** [comment]
> Not sure if it's in the scope of this PR, but I think that SafeSocketHandle should be moved from the common folder to System.Net.Sockets. Also, not sure if SafeCloseSocketAndEvent is used anywhere else, I think it can be removed.

I'll take a look. Thanks.

**Pull request merged**